mitosheet/mitosheet/api/get_defined_df_names.py

```diff
@@ -30,7 +30,7 @@ def __exit__(self, *args):
     with Capturing() as output:
         ipython.run_line_magic("who",  "DataFrame")
         
-    output = [line.strip() for line in output]
+    output = [df.strip() for line in output for df in line.split("\t") if df.strip() != '']
```

Comment:
It looks like you can instead do `who_ls` to get a list of the dataframe names. This would save us from this parsing. See here: https://stackoverflow.com/a/34950500/14993381

Reply:
For some reason, this isn't working in the context of this magic call! 

```
def get_df_names():
    from IPython import get_ipython
    from io import StringIO 
    import sys

    ipython = get_ipython()

    class Capturing(list):
        def __enter__(self):
            self._stdout = sys.stdout
            sys.stdout = self._stringio = StringIO()
            return self
        def __exit__(self, *args):
            self.extend(self._stringio.getvalue().splitlines())
            del self._stringio    # free up some memory
            sys.stdout = self._stdout

    with Capturing() as output:
        ipython.run_line_magic("who_ls",  "DataFrame")
                
    output = [line.strip() for line in output]
    return output

get_df_names()
```

Feel free to test, but gonna stick with the working approach for now.